src/node.js

```diff
@@ -390,6 +391,61 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var pendingRejections = [];
```

Comment:
It's not about optimizations but how iterator protocol is specified - every iteration allocates an object with done and value properties. However forEach could be interesting as V8 shares the iteration object between iterations.